actions/tahoe-lafs

```diff
@@ -0,0 +1,250 @@
+    intro_parser_add = subparsers.add_parser('add-introducer',
+                                             help="Add an introducer to the "
+                                                  "storage node's "
+                                                  "list of introducers.")
+    intro_parser_add.add_argument('--introducer', help="Add an introducer to "
+                                                       "the storage node's "
+                                                       "list of introducers "
+                                                       "Param introducer must "
+                                                       "be a tuple of "
+                                                       "(pet_name, furl)")
+    intro_parser_remove = subparsers.add_parser('remove-introducer',
+                                                help="Rename the introducer "
+                                                     "entry in the "
+                                                     "introducers.yaml file "
+                                                     "specified by the param")
+    intro_parser_remove.add_argument(
+        '--pet-name', help='The domain name that will be used by '
+                           'Matrix Synapse')
+    subparsers.add_parser('get-introducers',
+                          help="Return a dictionary of all introducers and "
+                               "their furls added to the storage node running "
+                               "on this FreedomBox.")
+    subparsers.add_parser('get-local-introducer',
+                          help="Return the name and furl of the introducer "
+                               "created on this FreedomBox")
+
+    return parser.parse_args()
+
+
+def subcommand_setup(arguments):
+    """Actions to be performed before installing Tahoe-LAFS."""
+    # Create tahoe group if needed.
+    try:
+        grp.getgrnam('tahoe-lafs')
+    except KeyError:
+        subprocess.run(['addgroup', 'tahoe-lafs'], check=True)
+
+    # Create tahoe user is needed.
+    try:
+        pwd.getpwnam('tahoe-lafs')
+    except KeyError:
+        subprocess.run(
+            [
+                'adduser', '--system', '--ingroup', 'tahoe-lafs',
+                '--home', '/var/lib/tahoe-lafs',
+                '--gecos', 'Tahoe-LAFS distributed file system', 'tahoe-lafs'
+            ],
+            check=True)
+
+    if not os.path.exists(tahoe_home):
+        os.makedirs(tahoe_home, mode=0o755)
+
+    shutil.chown(tahoe_home, user='tahoe-lafs', group='tahoe-lafs')
+
+    if not os.path.exists(domain_name_file):
+        with open(domain_name_file, 'w') as dnf:
+            dnf.write(arguments.domain_name)
+
+
+def subcommand_autostart(_):
+    """Automatically start all introducers and storage nodes on system startup.
+    """
+    with open('/etc/default/tahoe-lafs', 'a') as autostart_conf:
+        autostart_conf.write('AUTOSTART="all"')
```

Comment:
We should make this idempotent, by first checking if the file contains this line already.